plugins/core/Standard/qRANSAC_SD/qRANSAC_SD_Commands.h

```diff
@@ -0,0 +1,514 @@
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#                         COPYRIGHT: Chris S Brown                       #
+//#                                                                        #
+//##########################################################################
+
+#ifndef M3C2_PLUGIN_COMMANDS_HEADER
+#define M3C2_PLUGIN_COMMANDS_HEADER
+
+//CloudCompare
+#include "ccCommandLineInterface.h"
+#include <ccMesh.h>
+#include <ccGenericMesh.h>
+
+//Local
+#include "qRANSAC_SD.h"
+
+constexpr char COMMAND_RANSAC[] = "RANSAC";
+constexpr char EPSILON_ABSOLUTE[] = "EPSILON_ABSOLUTE";
+constexpr char EPSILON_PERCENTAGE_OF_SCALE[] = "EPSILON_PERCENTAGE_OF_SCALE";
+constexpr char BITMAP_EPSILON_PERCENTAGE_OF_SCALE[] = "BITMAP_EPSILON_PERCENTAGE_OF_SCALE";
+constexpr char BITMAP_EPSILON_ABSOLUTE[] = "BITMAP_EPSILON_ABSOLUTE";
+constexpr char SUPPORT_POINTS[] = "SUPPORT_POINTS";
+constexpr char MAX_NORMAL_DEV[] = "MAX_NORMAL_DEV";
+constexpr char PROBABILITY[] = "PROBABILITY";
+constexpr char ENABLE_PRIMITIVE[] = "ENABLE_PRIMITIVE";
+constexpr char OUT_CLOUD_DIR[] = "OUT_CLOUD_DIR";
+constexpr char OUT_MESH_DIR[] = "OUT_MESH_DIR";
+constexpr char OUT_PAIR_DIR[] = "OUT_PAIR_DIR";
+constexpr char OUT_GROUP_DIR[] = "OUT_GROUP_DIR";
+constexpr char OUTPUT_INDIVIDUAL_PRIMITIVES[] = "OUTPUT_INDIVIDUAL_PRIMITIVES";
+constexpr char OUTPUT_INDIVIDUAL_SUBCLOUDS[] = "OUTPUT_INDIVIDUAL_SUBCLOUDS";
+constexpr char OUTPUT_INDIVIDUAL_PAIRED_CLOUD_PRIMITIVE[] = "OUTPUT_INDIVIDUAL_PAIRED_CLOUD_PRIMITIVE";
+constexpr char OUTPUT_GROUPED[] = "OUTPUT_GROUPED";
+
+constexpr char PRIM_PLANE[] = "PLANE";
+constexpr char PRIM_SPHERE[] = "SPHERE";
+constexpr char PRIM_CYLINDER[] = "CYLINDER";
+constexpr char PRIM_CONE[] = "CONE";
+constexpr char PRIM_TORUS[] = "TORUS";
+
+
+struct CommandRANSAC : public ccCommandLineInterface::Command
+{
+	CommandRANSAC() : ccCommandLineInterface::Command("RANSAC", COMMAND_RANSAC) {}
+
+	virtual bool process(ccCommandLineInterface& cmd) override
+	{
+		if (cmd.clouds().empty())
+		{
+			return cmd.error(QObject::tr("No point cloud to attempt RANSAC on (be sure to open one with \"-O [cloud filename]\" before \"-%2\")").arg(COMMAND_RANSAC));
+		}
+		qRansacSD::RansacParams params;
+		QStringList paramNames = QStringList() << EPSILON_ABSOLUTE << EPSILON_PERCENTAGE_OF_SCALE <<
+			BITMAP_EPSILON_PERCENTAGE_OF_SCALE << BITMAP_EPSILON_ABSOLUTE <<
+			SUPPORT_POINTS << MAX_NORMAL_DEV << PROBABILITY << ENABLE_PRIMITIVE <<
+			OUT_CLOUD_DIR << OUT_MESH_DIR << OUT_GROUP_DIR << OUT_PAIR_DIR << OUTPUT_INDIVIDUAL_PRIMITIVES <<
+			OUTPUT_INDIVIDUAL_SUBCLOUDS << OUTPUT_GROUPED << OUTPUT_INDIVIDUAL_PAIRED_CLOUD_PRIMITIVE;
+		QStringList primitiveNames = QStringList() << PRIM_PLANE << PRIM_SPHERE << PRIM_CYLINDER << PRIM_CONE << PRIM_TORUS;
+		QString outputCloudsDir = "";
```

Comment:
Not sure about this initialization to "" (QStrings could be left to their default 'null' value?)

Reply:
yeah it worked without the initialization to "", I guess I'm just used to always initializing to something